server/sonar-server/src/test/java/org/sonar/server/computation/source/ReportIteratorTest.java

```diff
@@ -71,6 +71,15 @@ public void read_report() throws Exception {
     assertThat(sut.next().getLine()).isEqualTo(1);
   }
 
+  @Test
+  public void call_has_next_when_already_closed() throws Exception {
+    sut = new ReportIterator<>(file, BatchReport.Coverage.PARSER);
+    assertThat(sut.next().getLine()).isEqualTo(1);
+    assertThat(sut.hasNext()).isFalse();
+    // not fail
```

Comment:
remove comment and add info in the name method … like do_not_fail_when_calling_has_next_with_iterator_already_closed() or something similar